eras/shelley/impl/src/Cardano/Ledger/Shelley/UTxO.hs

```diff
@@ -395,7 +404,19 @@ keyRefunds pp tx = length deregistrations <×> getField @"_keyDeposit" pp
     deregistrations = filter isDeRegKey (toList $ tx ^. certsTxBodyL)
 
 class EraTxBody era => EraUTxO era where
+  -- | A customizable type on per era basis for the information required to find all
+  -- scripts needed for the transaction.
+  type ScriptsNeeded era :: Type
+
+  -- | Calculate all the value that is being consumed by the transaction.
   getConsumedValue :: PParams era -> UTxO era -> TxBody era -> Value era
 
+  -- | Produce all the information required for figuring out which scripts are required
+  -- for the transaction to be valid, once those scripts are evaluated
+  getScriptsNeeded :: UTxO era -> TxBody era -> ScriptsNeeded era
+
 instance Crypto c => EraUTxO (ShelleyEra c) where
+  type ScriptsNeeded (ShelleyEra c) = Set (ScriptHash c)
```

Comment:
should we use a newtype to name this better, ie `ShelleyScriptsNeeded`?

Reply:
Yeah, sure, why not. I'll make both ShelleyScriptsNeeded and AlonzoScriptsNeeded into the newtypes.

Reply:
thanks!

Reply:
Switched to `newtypes` and added `ShelleyScriptsNeeded`